### DISCUSSION TITLE
Change color of comment in code block

### DISCUSSION THREAD

**Discussion by blinkerfluessigkeit** [typst/typst#3018] [Q&A] [answered]
Hello guys,

I'm writing technical documentation for myself and I have set a color scheme that indicates for each line, whether it is a comment/terminal command/etc.

I was trying to apply a show rule that changes the color of each line of text that starts with a '#' to the desired rgb-value, but I couldn't get it working so far (do I need to learn Rust after all?)

Here's what I've tried so far:
```typ
#show raw: set text(fill: rgb("#bf8040")) if text.starts_with("#")
```
The error message I get is: "function `text` does not contain field `starts_with`"

**Comment by laurmaedje** [comment, marked as answer]
This would do the trick:
```typ
#show raw.line: it => {
  set text(fill: rgb("#bf8040")) if it.text.starts-with("#")
  it
}

Hello
# Comment
World
```
In your example, `text` is just the text function and not a specific string. Using a `raw.line` show rule, we can get access to the concrete string.

**Reply by blinkerfluessigkeit** [reply]
Thank you very much, that works perfectly :)

**Reply by blinkerfluessigkeit** [reply]
Hey, do you think it would also be possible to change the color for inline comments? For example in this:
```
Text        # Inline comment
```
the line doesn't start with a '#'-symbol, but I'd like the comment (including the '#') to still be colored.

**Reply by laurmaedje** [reply]
You could probably do it with a regex show rule inside of your raw show rule.

**Reply by blinkerfluessigkeit** [reply]
That was exactly the hint I needed, thanks. For anyone who might read and need this in the future, here's my updated show rule:

```typst
#let color_comment = rgb("#bf8040")

#show raw.line: it => {
    show regex(" *#.*"): it => {
        set text(fill: color_comment)
        it
    }
    it
}
```
